syntax = "proto3";

package phoenix.game;

import "google/protobuf/empty.proto";
import "phoenixapi/protos/game/entities.proto";
import "phoenixapi/protos/position.proto";

service PlayerManager {
    rpc GetPlayerObjManager(google.protobuf.Empty) returns (PlayerObjManager) {}

    rpc ResetPlayerState(google.protobuf.Empty) returns (google.protobuf.Empty) {}

    rpc Walk(Position) returns (google.protobuf.Empty) {}
}

message PlayerObjManager {
    Position position = 1;
    Position dest_position = 2;
    int32 state = 5; // temp as int32
    entities.Player player = 6; // your character player data
    sint32 id = 7;
    bool is_resting = 8;
}